### DISCUSSION TITLE
How to Integrate Voice Search in Android tv and ios tv

### DISCUSSION THREAD

**Discussion by GitHubShasiKumar** [react-native-tvos/react-native-tvos#677] [Q&A]
### Description

Hi @douglowder  ,This is not a kind of bug ,its more of kind of question , is there any possible way to interate voice search (text to speech using remote) in react native tvos for Android Tv and Ios Tv , Please Provide Some Input Thanks 

### React Native Version

npm:react-native-tvos@0.69.8-0

### Output of `npx react-native info`

System:
OS: macOS 13.5
CPU: (8) arm64 Apple M1
Memory: 97.75 MB / 8.00 GB
Shell: 5.9 - /bin/zsh
Binaries:
Node: 20.3.1 - /opt/homebrew/bin/node
Yarn: 1.22.19 - /opt/homebrew/bin/yarn
npm: 9.6.7 - /opt/homebrew/bin/npm
Watchman: 2023.07.03.00 - /opt/homebrew/bin/watchman
Managers:
CocoaPods: 1.12.1 - /opt/homebrew/bin/pod
SDKs:
iOS SDK:
Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, watchOS 10.2
Android SDK: Not Found
IDEs:
Android Studio: 2022.3 AI-223.8836.35.2231.11005911
Xcode: 15.1/15C65 - /usr/bin/xcodebuild
Languages:
Java: 19.0.2 - /usr/bin/javac
npmPackages:
@react-native-community/cli: Not Found
react: 18.0.0 => 18.0.0
react-native: Not Found
react-native-macos: Not Found
react-native-tvos: 0.69.8-0
npmGlobalPackages:
react-native: Not Found

### Steps to reproduce

NA

### Snack, screenshot, or link to a repository

NA

**Comment by GitHubShasiKumar** [comment]
Hi @douglowder Currently iam using @react-native-voice/voice latest version its bit working fine in android tv but in Apple tv it is not working

**Comment by douglowder** [comment]
I looked at the react-native-voice repo... last release was 2 years ago, and its podspec is configured for iOS only, tvOS is not enabled.

**Comment by GitHubShasiKumar** [comment]
Hi @douglowder  thanks for the response, so is there any key to trigger the voice recognizer programically in  Android Tv 
 without using the package for the textinput and without opening keyboard 
 like  the audio button in youtube
 
![Screenshot_1707827128](https://github.com/react-native-tvos/react-native-tvos/assets/107665648/49ba5c64-3125-4930-b938-a16ea2cd214c)

**Comment by chourouk-mrad** [comment]
Hello @douglowder 
I'm trying to implement the speech-to-text feature on appleTV, is there any recommandation for this ? 
We have tried to use apple native module Speech but it's not available on tvos .

**Comment by kashyap-rathod-us** [comment]
Hello @douglowder 
Any update on this feature for tvos ?

**Reply by ArtyomKulik** [reply]
Expo Audio doesn't work? @kashyap-rathod-us
